File: MDVision.IEHR/Controls/Admin/Admin_PasswordConfiguration.js
Admin_PasswordConfiguration = {
    bIsFirstLoad: true,
    Load: function (params) {
        var self = $('#frmPasswordConfiguration');
        if (Admin_PasswordConfiguration.bIsFirstLoad) {
            Admin_PasswordConfiguration.bIsFirstLoad = false;
            self.loadDropDowns(true).done(function () {
                //serialize Data.
                $('#frmPasswordConfiguration').data('serialize', $('#frmPasswordConfiguration').serialize());
                Admin_PasswordConfiguration.ValidatePasswordConfiguration();

                Admin_PasswordConfiguration.EntityChanged($("#frmPasswordConfiguration #lstEntityId"));
            });
        }
    },

    PasswordConfigurationFill: function (PasswordConfigurationID) {
        Admin_PasswordConfiguration.FillPasswordConfiguration(PasswordConfigurationID).done(function (response) {
            if (response.status != false) {

                var Admin_PasswordConfigurationData = JSON.parse(response.PasswordConfigurationFill_JSON);
                var self = $("#frmPasswordConfiguration");
                utility.bindMyJSON(true, Admin_PasswordConfigurationData, false, self).done(function () {
                    //serialize Data.
                    $('#frmPasswordConfiguration').data('serialize', $('#frmPasswordConfiguration').serialize());
                });
            }
            else {
                Admin_PasswordConfiguration.setDefaultConfiguration();
                //                utility.DisplayMessages(response.Message, 3);
            }
        });
    },

    FillPasswordConfiguration: function (PasswordConfigurationID) {
        var data = "PasswordConfigurationID=" + PasswordConfigurationID;
        // serach parameter , class name, command name of class 
        return MDVisionService.defaultService(data, "ADMIN_PASSWORD_CONFIGURATION", "FILL_PASSWORD_CONFIGURATION");
    },

    ValidatePasswordConfiguration: function () {
        $('#frmPasswordConfiguration')
            .bootstrapValidator({
                live: 'disabled',
                message: 'This value is not valid',
                feedbackIcons: {
                    valid: 'glyphicon glyphicon-ok',
                    invalid: 'glyphicon glyphicon-remove',
                    validating: 'glyphicon glyphicon-refresh'
                },
                fields: {

                    Entity: {
                        group: '.col-sm-2',
                        validators: {
                            notEmpty: {
                                message: ''
                            }
                        }
                    },

                    MinPasswordLength: {
                        group: '.col-sm-2',
                        validators: {
                            notEmpty: {
                                message: ''
                            }
                        }
                    },
                    MinSpecialCharacter: {
                        group: '.col-sm-2',
                        validators: {
                            notEmpty: {
                                message: ''
                            }
                        }
                    },
                    MinAlphaCharacter: {
                        group: '.col-sm-2',
                        validators: {
                            notEmpty: {
                                message: ''
                            }
                        }
                    },
                    MinNumericCharacter: {
                        group: '.col-sm-2',
                        validators: {
                            notEmpty: {
                                message: ''
                            }
                        }
                    },
                    MinUppercaseCharacter: {
                        group: '.col-sm-2',
                        validators: {
                            notEmpty: {
                                message: ''
                            }
                        }
                    },
                    MaxPasswordAge: {
                        group: '.col-sm-2',
                        validators: {
                            notEmpty: {
                                message: ''
                            }
                        }
                    },
                    PasswordHistory: {
                        group: '.col-sm-2',
                        validators: {
                            notEmpty: {
                                message: ''
                            }
                        }
                    },

                    AccountLockoutThreshold: {
                        group: '.col-sm-2',
                        validators: {
                            notEmpty: {
                                message: ''
                            }
                        }
                    },

                    IdleSessionTimeout: {
                        group: '.col-sm-2',
                        validators: {
                            notEmpty: {
                                message: ''
                            }
                        }
                    },

                }
            })
        .on('success.form.bv', function (e) {
            e.preventDefault();
            Admin_PasswordConfiguration.PasswordConfigurationUdpate();
        });
    },

    PasswordConfigurationUdpate: function () {
        var self = $("#frmPasswordConfiguration");
        var myJSON = self.getMyJSON();
        //password configuration id and entityid would be the same
        var EntityId = $("#frmPasswordConfiguration #lstEntityId").val();
        if (EntityId != "") {
            Admin_PasswordConfiguration.UpdatePasswordConfiguration(myJSON, EntityId).done(function (response) {
                if (response.status != false) {
                    utility.DisplayMessages(response.Message, 1);
                }
                else {
                    utility.DisplayMessages(response.Message, 3);
                }
            });
        }
    },

    UpdatePasswordConfiguration: function (PasswordConfigurationData, PasswordConfigurationID) {
        var data = "PasswordConfigurationData=" + PasswordConfigurationData + "&PasswordConfigurationID=" + PasswordConfigurationID;
        // serach parameter , class name, command name of class 
        return MDVisionService.defaultService(data, "ADMIN_PASSWORD_CONFIGURATION", "UPDATE_PASSWORD_CONFIGURATION");
    },

    EntityChanged: function (Ctrl) {

        var EntityId = $(Ctrl).val();
        if (EntityId != "") {
            Admin_PasswordConfiguration.PasswordConfigurationFill(EntityId);
        } else {


        }
    },

    setDefaultConfiguration: function () {
        $("#frmPasswordConfiguration #txtMinPasswordLength").val(8);
        $("#frmPasswordConfiguration #txtMinSpecialCharacter").val(1);
        $("#frmPasswordConfiguration #txtMinAlphaCharacter").val(1);
        $("#frmPasswordConfiguration #txtMinNumericCharacter").val(1);
        $("#frmPasswordConfiguration #txtMinUppercaseCharacter").val(1);
        $("#frmPasswordConfiguration #txtMaxPasswordAge").val(60);
        $("#frmPasswordConfiguration #txtPasswordHistory").val(24);
        $("#frmPasswordConfiguration #txtAccountLockoutThreshold").val(0);
        $("#frmPasswordConfiguration #txtIdleSessionTimeout").val(10);
    },


    UnLoadTab: function (Tab) {
        RemoveAdminTab(Tab);
    },
}

File: MDVision.IEHR/Controls/Admin/Admin_PasswordConfiguration.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using MDVision.Datasets;
using MDVision.Business.BCommon;
using MDVision.Common.Utilities;
using MDVision.Business.BLL;
using MDVision.Common.Shared;
using System.Data;
using MDVision.IEHR.Common;
using System.Web.Script.Serialization;
using Newtonsoft.Json;

namespace MDVision.IEHR.Controls.Admin
{
    public class Admin_PasswordConfiguration
    {
        private BLLAdminSecurity BLLAdminSecurityObj = null;
        public Admin_PasswordConfiguration()
        {
            BLLAdminSecurityObj = new BLLAdminSecurity();
        }

        #region Singleton
        private static Admin_PasswordConfiguration _obj = null;
        public static Admin_PasswordConfiguration Instance()
        {
            if (_obj == null)
                _obj = new Admin_PasswordConfiguration();
            return _obj;
        }
        #endregion

        #region Private Functions


        /// <summary>
        /// 
        /// </summary>
        /// <param name="fieldsJSON"></param>
        /// <param name="PasswordConfigurationID"></param>
        /// <returns></returns>
        private string SaveUpdatePasswordConfiguration(string fieldsJSON)
        {
            try
            {
                int minPasswordLength = 8;
                int minSpecialCharacter = 0;
                int minAlphaCharacter = 0;
                int minNumeriCharacter = 0;
                int minUppercaseeCharacter = 0;

                JavaScriptSerializer ser = new JavaScriptSerializer();
                var SearchedfieldsJSON = ser.Deserialize<dynamic>(fieldsJSON);

                DSUsers dsUser = new DSUsers();
                DSUsers.PasswordConfigurationRow dr = dsUser.PasswordConfiguration.NewPasswordConfigurationRow();

                if (!string.IsNullOrEmpty(SearchedfieldsJSON["lstEntityId"]))
                {
                    dr.PasswordConfigurationId = MDVUtility.ToInt64(SearchedfieldsJSON["lstEntityId"]);
                }
                else
                {
                    throw new Exception("please select an Entity First");
                }

                minPasswordLength = MDVUtility.ToInt32(SearchedfieldsJSON["txtMinPasswordLength"]);
                minSpecialCharacter = MDVUtility.ToInt32(SearchedfieldsJSON["txtMinAlphaCharacter"]);
                minAlphaCharacter = MDVUtility.ToInt32(SearchedfieldsJSON["txtMinNumericCharacter"]);
                minNumeriCharacter = MDVUtility.ToInt32(SearchedfieldsJSON["txtMinUppercaseCharacter"]);
                minUppercaseeCharacter = MDVUtility.ToInt32(SearchedfieldsJSON["txtMinUppercaseCharacter"]);
              
                dr.MinPasswordLength = minPasswordLength;
                dr.MinSpecialCharacter = minSpecialCharacter;
                dr.MinAlphaCharacter = minAlphaCharacter;
                dr.MinNumericCharacter = minNumeriCharacter;
                dr.MinUppercaseCharacter = minUppercaseeCharacter;
                dr.MaxPasswordAge = MDVUtility.ToInt32(SearchedfieldsJSON["txtMaxPasswordAge"]);
                dr.PasswordHistory = MDVUtility.ToInt32(SearchedfieldsJSON["txtPasswordHistory"]);
                dr.AccountLockoutThreshold = MDVUtility.ToInt32(SearchedfieldsJSON["txtAccountLockoutThreshold"]);
                dr.IdleSessionTimeout = MDVUtility.ToInt32(SearchedfieldsJSON["txtIdleSessionTimeout"]);
                dr.PasswordRegex = MDVUtility.GenerateRegex(minPasswordLength, minSpecialCharacter, minAlphaCharacter, minNumeriCharacter, minUppercaseeCharacter);
                dr.IsActive = true;
                dr.CreatedBy = MDVUtility.DecryptFrom64(MDVSession.Current.AppUserName);
                dr.CreatedOn = DateTime.Now;
                dr.ModifiedBy = MDVUtility.DecryptFrom64(MDVSession.Current.AppUserName);
                dr.ModifiedOn = DateTime.Now;
                #region Database Updation
                dsUser.PasswordConfiguration.AddPasswordConfigurationRow(dr);
                //  dsUser.Users.AcceptChanges();

                if (dsUser.PasswordConfiguration.Rows.Count > 0)
                {
                    //    dsUser.PasswordConfiguration.Rows[0].SetModified();
                    BLObject<DSUsers> obj = BLLAdminSecurityObj.SaveUpdatePasswordConfiguration(ref dsUser);
                    if (obj.Data != null)
                    {
                        var response = new
                        {
                            status = true,
                            Message = Common.AppPrivileges.Save_Message
                        };
                        return (JsonConvert.SerializeObject(response));
                    }
                    else
                    {
                        var response = new
                        {
                            status = false,
                            Message = obj.Message
                        };
                        return JsonConvert.SerializeObject(response);
                    }
                }
                else
                {
                    var response = new
                    {
                        status = false,
                        Message = ""
                    };
                    return JsonConvert.SerializeObject(response);
                }
                #endregion
            }
            catch (Exception ex)
            {
                var response = new
                {
                    status = false,
                    Message = MDVCustomException.HumanReadableMessage(ex.Message)
                };
                return (JsonConvert.SerializeObject(response));
            }
        }

        /// <summary>
        /// 
        /// </summary>
        /// <param name="PasswordConfigurationID"></param>
        /// <returns></returns>
        private string FillPasswordConfiguration(Int64 PasswordConfigurationID)
        {
            try
            {
                if (string.IsNullOrEmpty(MDVUtility.ToStr(PasswordConfigurationID)))
                {
                    var response = new
                    {
                        status = false,
                        Message = MDVUtility.ToStr(Common.AppPrivileges.Select_Message)
                    };
                    return JsonConvert.SerializeObject(response);
                }
                else
                {
                    DSUsers dsUsers = null;
                    BLObject<DSUsers> objPasswordConfiguration = BLLAdminSecurityObj.LoadPasswordConfiguration(PasswordConfigurationID);
                    if (objPasswordConfiguration.Data != null)
                    {
                        dsUsers = objPasswordConfiguration.Data;
                        if (dsUsers.PasswordConfiguration.Rows.Count > 0)
                        {
                            DSUsers.PasswordConfigurationRow dr = (DSUsers.PasswordConfigurationRow)dsUsers.PasswordConfiguration.Rows[0];

                            var keyValues = new Dictionary<string, string>
                        {
                           // { "lstEntityId", MDVUtility.ToStr(dr[dsUsers.PasswordConfiguration.PasswordConfigurationIdColumn.ColumnName])},
                            { "txtMinPasswordLength", MDVUtility.ToStr(dr.MinPasswordLength)},
                            { "txtMinSpecialCharacter", MDVUtility.ToStr(dr.MinSpecialCharacter)},
                            { "txtMinAlphaCharacter", MDVUtility.ToStr(dr.MinAlphaCharacter)},
                            { "txtMinNumericCharacter", MDVUtility.ToStr(dr.MinNumericCharacter)},
                            { "txtMinUppercaseCharacter", MDVUtility.ToStr(dr.MinUppercaseCharacter)},
                            { "txtMaxPasswordAge", MDVUtility.ToStr(dr.MaxPasswordAge)},
                            { "txtPasswordHistory", MDVUtility.ToStr(dr.PasswordHistory)},
                            { "txtAccountLockoutThreshold", MDVUtility.ToStr(dr.AccountLockoutThreshold)},
                            { "txtIdleSessionTimeout", MDVUtility.ToStr(dr.IdleSessionTimeout)},
                            
                        };
                            JavaScriptSerializer js = new JavaScriptSerializer();
                            var response = new
                            {
                                status = true,
                                PasswordConfigurationFill_JSON = js.Serialize(keyValues)
                            };
                            return (JsonConvert.SerializeObject(response));
                        }
                        else
                        {
                            var response = new
                            {
                                status = false,
                                Message = AppPrivileges.No_Record_Message,
                            };
                            return (JsonConvert.SerializeObject(response));
                        }
                    }
                    else
                    {
                        var response = new
                        {
                            status = false,
                            Message = objPasswordConfiguration.Message
                        };
                        return (JsonConvert.SerializeObject(response));
                    }

                }

            }
            catch (Exception ex)
            {
                var response = new
                {
                    status = false,
                    Message = MDVCustomException.HumanReadableMessage(ex.Message),
                };
                return (JsonConvert.SerializeObject(response));
            }
        }


        #endregion

        #region Service Command Handler

        public void CommandHandler(HttpContext context)
        {
            string cammandAction = context.Request.QueryString["cammandAction"].ToUpper();

            switch (cammandAction)
            {
                case "FILL_PASSWORD_CONFIGURATION":
                    {
                        string strPasswordConfigurationId = context.Request["PasswordConfigurationID"];
                        string strJSONData = FillPasswordConfiguration(MDVUtility.ToInt64(strPasswordConfigurationId));
                        context.Response.ContentType = "text/plain";
                        context.Response.Write(strJSONData);
                    }
                    break;

                case "UPDATE_PASSWORD_CONFIGURATION":
                    {
                        string fieldsJSON = context.Request["PasswordConfigurationData"];
                        string strJSONData = SaveUpdatePasswordConfiguration(fieldsJSON);

                        context.Response.ContentType = "text/plain";
                        context.Response.Write(strJSONData);
                    }
                    break;
            }
        }
        #endregion
    }
}